extmod/uasyncio/core.py

```diff
@@ -215,20 +234,41 @@ def run(coro):
 
 
 class Loop:
-    def create_task(self, coro):
+    @staticmethod
+    def create_task(coro):
         return create_task(coro)
 
-    def run_forever(self):
+    @staticmethod
+    def run_forever():
         run_until_complete()
         # TODO should keep running until .stop() is called, even if there're no tasks left
 
-    def run_until_complete(self, aw):
+    @staticmethod
+    def run_until_complete(aw):
         return run_until_complete(_promote_to_task(aw))
 
-    def close(self):
+    @staticmethod
+    def close():
         pass
 
+    @staticmethod
+    def set_exception_handler(handler):
+        global exc_handler
+        exc_handler = handler
+
+    @staticmethod
+    def get_exception_handler():
+        return exc_handler
+
+    @classmethod
+    def default_exception_handler(cls, context):
+        exc_handler(cls, context)
+
+    @classmethod
+    def call_exception_handler(cls, context):
+        exc_handler(cls, context)
```

Comment:
As far as I can tell from the CPython docs this should be `_exc_handler`

Reply:
Where did you read that? 
"exc_handler" isn't exposed as part of the API. In fact, I should have named it differently to reflect that it is only an internal variable not to be used or changed externally.

Reply:
Hey, I have no experience with this exception handling, so it's very possible that I'm misunderstanding. What I saw on the page you linked to is:

>loop.default_exception_handler(context)¶
>Default exception handler.
>This is called when an exception occurs and no exception handler is set. This can be called by a custom exception handler that wants to defer to the default handler behavior.

My understanding of this last sentence is that if I register a custom exception handler (thus `exc_handler == my_custom_handler`) and in this custom handler I want to just call the standard default handler I can call `loop.default_exception_handler()`, thus the latter should call `_exc_handler`. (More elegant would be to rename `_exc_handler` to `default_exception_handler`.) But maybe I'm not understanding it quite right (I find the python docs more ambiguous than I'm used to for this type of stuff...)

Reply:
Whooops! Now I see my mistake, my comment should have been attached to line 265 (4 lines up).

Reply:
Ah, indeed. That is a mistake on my part. it should be _exec_handler here.